src/main/java/org/opensearch/knn/index/MethodComponent.java

```diff
@@ -82,22 +84,39 @@ public String getName() {
      * Validate that the methodComponentContext is a valid configuration for this methodComponent
      *
      * @param methodComponentContext to be validated
+     * @return ValidationException produced by validation errors; null if no validations errors.
      */
-    public void validate(MethodComponentContext methodComponentContext) {
+    public ValidationException validate(MethodComponentContext methodComponentContext) {
```

Comment:
I have been thinking about this for couple of days. Do we have to return exception instead of throw exception.

Reply:
Yes, because ValidationExceptions need to be aggregated together. There are a few examples of this in OpenSearch. [Here](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/action/ValidateActions.java#L37) is one

Reply:
I see. It definitely looks like anti-pattern compared to typical java applications.